File: solutions/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]: 
        # n = len(nums)
        # for i in range(n):
        #     for j in range(i+1, n):
        #         if nums[i]+nums[j] == target:
        #             return [i,j]
        num_map = {}
        for i, num in enumerate(nums):
            comp = target - num
            if comp in num_map:
                return[num_map[comp],i]
            num_map[num] = i
        return [-1,-1]

File: solutions/0001-two-sum/solution.java
class Solution 
{
    public int[] twoSum(int[] nums, int target) 
    {
        for(int i =1; i<nums.length; i++)
        {
            for(int j = i; j<nums.length; j++)
            {
                if(nums[j-i] + nums[j] == target) return new int[]{j-i,j};
                // {
                //     arr[0] = i;
                //     arr[1] = j;
                //     break;
                // }
            }
        }
        return new int[]{};
    }
}
